File: platform/web/packages/components/src/roles.ts

export const userAdminRoles = [
    "tr_orchestrator_admin",
    "tr_project_manager_admin",
    "tr_stakeholder_admin"
] as const

export const userBaseRoles = [
    "tr_orchestrator_user",
    "tr_project_manager_user",
    "tr_stakeholder_user",
] as const

export const userRoles = [
    "super_admin",
    ...userAdminRoles,
    ...userBaseRoles
] as const

export const orgRoles = [
    "tr_orchestrator",
    "tr_project_manager",
    "tr_stakeholder"
] as const


export const userEffectiveRoles = [...userRoles, ...orgRoles]
export type Roles = typeof userEffectiveRoles[number]

File: platform/commons/src/commonMain/kotlin/io/komune/registry/s2/commons/auth/Roles.kt
package io.komune.registry.s2.commons.auth

import kotlin.js.JsExport
import kotlin.js.JsName

@JsExport
@JsName("Roles")
@Deprecated("Will be removed, use Permissions instead.")
object Roles {
    const val ORCHESTRATOR = "tr_orchestrator"
    const val ORCHESTRATOR_ADMIN = "tr_orchestrator_admin"
    const val ORCHESTRATOR_USER = "tr_orchestrator_user"

    const val PROJECT_MANAGER = "tr_project_manager"
    const val PROJECT_MANAGER_ADMIN = "tr_project_manager_admin"
    const val PROJECT_MANAGER_USER = "tr_project_manager_user"

    const val STAKEHOLDER = "tr_stakeholder"
    const val STAKEHOLDER_ADMIN = "tr_stakeholder_admin"
    const val STAKEHOLDER_USER = "tr_stakeholder_user"
}
